app/models/user.rb

```diff
@@ -80,11 +80,20 @@ def as_json_for_event_api
   end
 
   def as_payload
-    as_json(only: %i[uid email role level state])
+    as_json(only: %i[uid email referral_id role level state])
   end
 
   private
 
+  def validate_and_assign_ref_id
```

Comment:
I disagree with the approach
Referral ID should be simply like email is set and saved, validated.
but there is no relationship between generating a UID and setting a referral ID.
we just need to validate it's correct format, and it exist in db.